### PR TITLE
[WIP] port to http master

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,3 @@
 julia 0.7
 MbedTLS
-HTTP 0.6.14 0.6.15
-
+HTTP
```
[the remark this answers is not in the document]
Comment (SimonDanisch) [reply REQUIRE]: master

File: src/HTTP.jl [hunk 2]
```diff
@@ -301,42 +322,46 @@ After a suspected connection task failure:
     end
 ```
 """
-function serve(server::ServerWS{T, H, W}, host, port, verbose) where {T, H, W}
-
+function serve(server::ServerWS{T, H, W}, host, port, verbose, sleeptime = 0.01) where {T, H, W}
```
Comment (hustf) [inline src/HTTP.jl]: Good idea to include sleeptime. We used to completely avoid keyword arguments in earlier Julia versions, but if somebody finds the time to incorporate it, there are only upsides now.

File: src/HTTP.jl [hunk 3]
```diff
@@ -301,42 +322,46 @@ After a suspected connection task failure:
...
+                HTTP.handle(server.handler.func, stream)
             end
+        catch err
+            put!(server.out, err)
+            put!(server.out, stacktrace(catch_backtrace()))
+        end
+    end
     return
 end
+
```
Comment (hustf) [inline src/HTTP.jl]: This is also changed in the http0.7.0 branch, which I suggest we merge first.

**Comment by hustf** [comment]
Thanks for the pull request and sorry for the late response. I had an overload with making 'syntax' changes to this package and took a small break. It's my fault of course that the tests (and benchmarks) used to be ugly, imperative scripts which has required a lot of rewriting. 

This pull request makes some nice changes, which we could of course merge to master at once.

But when it comes to interfaces we should take small steps. This PR skips HTTP versions. 

Merging this later will be easier when including the improvements on branch 'origin/http0.7.0', since it addresses things like HTTP.kill and improves the tests.

**Comment by SimonDanisch** [comment]
> http0.7.0 branch 

Oh I didn't see that one... probably worth making a PR?

**Comment by hustf** [comment]
I'll make a PR of it later, but I wanted to make it work first, throw out temporary debug code and restructure the tests.

Unfortunately, I met with difficulty. Simple HTTP.requests aren't working at the moment. That's probably because I didn't grok the new interface. I'll make a commit with the latest temporary work now.

**Comment by hustf** [comment]
Ok, this interface change got me stuck for hours. I should have checked in HTTP at once once of course, and I'm putting it here in case it's related to the failing tests for this PR or in other tests you may be working with.

Julia 0.6.4, i.e. HTTP version 0.6.14
```
using HTTP
begin
           r = HTTP.request("GET", "http://www.nrk.no")
           println( "NRK Status is ", r.status)
end
NRK Status is 200
```
Julia 0.7, i.e. HTTP version 0.7.1
```
ERROR: LoadError: DNSError: ttpbin.org, unknown node or service (EAI_NONAME)
```
The error was output after making a successful post request first. On a second try, it just locks up for a long time. I guess the function needs more arguments now. Will fix another day...

**Comment by hustf** [comment]
Today, a new version of MBedTLS was released. It should only affect error messages, but anyhow the strange errors in HTTP.request on my local windows computer went away after MbedTLS triggered rebuilding HTTP. I have good hope that the remaining tests will pass when the debug cleanup is finished.

**Comment by hustf** [comment]
We never closed this because it still contains unimplemented improvements. 
Just making a note of considering including sleeptime = 0.01 as a default (if we'll still keep these constructs inside of this package, and the layout improvements.

**Comment by hustf** [comment]
I think we implemented these ideas as far as they are still relevant.
There's still no nice way to close servers when using the 'listen' syntax, but close(ServerWS) works smoothly.
Closing.

**Pull request closed without merging**